### DISCUSSION TITLE
the admin user in config.yaml , is it a must ?

### DISCUSSION THREAD

**Discussion by thomasdba** [zitadel/zitadel#6526] [General]
From Doc , it points :
The admin user is only needed for the installation step but can also be avoided by preparing the necessary requirements:

the zitadel user
the zitadel database
granting required permissions to the zitadel user
CREATE ROLE zitadel LOGIN;
CREATE DATABASE zitadel;
GRANT CONNECT, CREATE ON DATABASE zitadel TO zitadel;

But if admin user part is missing in config , zitadel cannot start . It is not a good practice to give admin user to zitadel , for most of cases , DBA can create user and database and grant owner to zitadel db .   There is no way to tell zitadel to bypass this .  It is fine to give user/dba to create role/db manually .

**Comment by ignisf** [comment]
The documentation is blatantly wrong on this IMO. I also second the opinion that it's a bad practice to have an admin account available to the application. Even provided you limit the secret to just an init/setup step, it is crying to be exploited in a supply chain attack, and frankly such a requirement is very uncharacteristic for a project that values security.

**Reply by adlerhurst** [reply]
how do you try to start zitadel? `zitadel start` and `zitadel setup` should work without an admin user

**Reply by ignisf** [reply]
These two steps do work without an admin user. However:

1. The documentation is wrong and the listed queries are of no help at all:
   1. They do not remove the requirement to run the init step as admin -- it fails.
   2. They are not enough to run the setup step -- it fails, too.
   3. You *must run the init step* and you must run it *with an admin user*. There is no way around that.
2. The documentation creates the impression (stating that it's idempotent and safe to run multiple times), and the source code for the init step strengthens it (as it is quite non-trivial, and thus completely expected to change), that you cannot just go ahead and skip the init step after running it just once before first use. It might change and you might miss necessary changes for an upgrade. So you *may run it as a step before starting the app up* and absolutely *must run it after each upgrade*.

Thus, while running without an admin user is possible, it creates the impression to not be a scenario that is tested/validated, or endorsed by the project.

**Reply by ignisf** [reply]
Also, this is literally the only rough edge I've found setting up the project. Everything else was smooth out of the box :clap:.

Cleaning up the docs and stating clearly that running just the init step as db admin is a must after pulling a new version would be sufficient to ease the setup process. Ideally to alleviate any tinfoil hat sysadmin concerns, the init step could be re-evaluated and any steps in it that require db admin could be made optional if already executed, and if not executed and no db admin permissions, they should fail with a clear 'needs admin, check docs message'. This also entails incorporating re-running init without db admin in CI so that no incidental breakage occurs.
